hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -410,8 +411,9 @@ public static Map<String, HoodieData<HoodieRecord>> convertMetadataToRecords(Hoo
       checkState(MetadataPartitionType.COLUMN_STATS.isMetadataPartitionAvailable(dataMetaClient),
           "Column stats partition must be enabled to generate partition stats. Please enable: " + HoodieMetadataConfig.ENABLE_METADATA_INDEX_COLUMN_STATS.key());
       // Generate Hoodie Pair data of partition name and list of column range metadata for all the files in that partition
+      boolean isDeletePartition = commitMetadata.getOperationType().equals(WriteOperationType.DELETE_PARTITION);
```

Comment:
As a follow-up, should we consider making this check general and make other index types also consider the `DELETE_PARTITION` write operation in a distinct way?

Reply:
Agree, created HUDI-8714 as followup. I think we can land this PR as files and partition stats are used in partition pruning so these indexes need to be strongly consistent. Column stats can be eventually consistent (update when replaced filegroups are cleaned) as colstats is used in data skipping after partition pruning.